pkg/common/utils.go

```diff
@@ -64,6 +65,18 @@ func VolumeIDToKey(id string) (*meta.Key, error) {
 	}
 }
 
+func SnapshotIDToKey(id string) (string, error) {
+	splitId := strings.Split(id, "/")
+	if len(splitId) != snapshotTotalElements {
+		return "", fmt.Errorf("failed to get id components. Expected projects/{project}/global/snapshot/{name}. Got: %s", id)
+	}
+	if splitId[volIDToplogyKey] == "global" {
+		return splitId[snapshotTotalElements-1],  nil
+	} else {
+		return "", fmt.Errorf("could not get id components, expected either zones or regions, got: %v", splitId[volIDToplogyKey])
```

Comment:
nit: expected global, not `volIDTopologyKey`

Reply:
I used volIDTopologyKey here because it would be same for snapshot. I can add a new one for snapshotIDTopologyKey too.